src/workflow/StreamlitUI.py

```diff
@@ -129,35 +164,29 @@ def upload_widget(
                         my_bar = st.progress(0)
                         for i, f in enumerate(files):
                             my_bar.progress((i + 1) / len(files))
-                            if not use_symlink:
+                            if use_copy:
                                 if os.path.isfile(f):
                                     shutil.copy(f, Path(files_dir, f.name))
                                 elif os.path.isdir(f):
                                     shutil.copytree(f, Path(files_dir, f.name), dirs_exist_ok=True)
                             else:
-                                symlink_target = Path(files_dir, f.name)
-                                if OS_PLATFORM == "win32":
-                                    # Detect if it's a file or directory
-                                    if f.is_file():
-                                        if not symlink_target.exists():
-                                            os.link(f, symlink_target)
-                                    else:
-                                        # Use mklink /J for directories
-                                        if not symlink_target.exists():
-                                            subprocess.call(["mklink", "/J", str(symlink_target), str(f)], shell=True)
-                                else:
-                                    if not symlink_target.exists():
-                                        os.symlink(f, symlink_target)
+                                # Do we need to change the reference of Path(st.session_state.workspace, "mzML-files") to point to local dir? 
+                                pass
```

Comment:
@t0mdavid-m if the `use_copy` is `False` (`Make a copy of files` deselected), how should we use the local directory path? 

Do we need to overwrite the `Path(st.session_state.workspace, "mzML-files")`, or should we add another session state var to indicate to use the local data files path, and then the app developer would just have to check that var to decide whether to use `Path(st.session_state.workspace, "mzML-files")` or `st.session_state.local_dir`?

Reply:
if we just update `self.params["mzML-files"]` with `st.session_state.local_dir` will `FileManager.get_files()` and `StreamlitUI.select_input_file()` and the rest work as expected?

Reply:
`StreamlitUI.select_input_file()` detects the files that were copied into the respective subdirectory ('mzML-files` in this case). Of course with the "symlink" this will not work as the files are not copied.

Thus I would suggest storing the filepaths in an external file within the respective directory. We could simply call it `external_files` and it should contain all paths to files that were not copied. `StreamlitUI.select_input_file()` coudl then read this file and append the paths to `files`. Everything downstream should then work as expected.

Reply:
Also, on second thought by providing the copy of files button with the warning, I think its okay to shift some responsibility to the user if they choose to disable it. `StreamlitUI.select_input_file()` could check if the local files still exist before providing them as possible inputs. Arguably, if the user decides to delete input files to their workflow while running it that is on them.